addons/nodelocaldns/nodelocaldns.yaml

```diff
@@ -0,0 +1,196 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: node-local-dns
+  namespace: kube-system
+  labels:
+    addonmanager.kubernetes.io/mode: Reconcile
+    kubeone.io/component: nodelocaldns
+data:
+  Corefile: |
+    {{ .Config.ClusterNetwork.ServiceDomainName }}:53 {
+      errors
+      cache {
+        success 9984 30
+        denial 9984 5
+      }
+      reload
+      loop
+      bind __PILLAR__LOCAL__DNS__
+      forward . __PILLAR__CLUSTER__DNS__ {
+        force_tcp
+      }
+      prometheus :9253
+      health __PILLAR__LOCAL__DNS__:8080
+    }
+    in-addr.arpa:53 {
+      errors
+      cache 30
+      reload
+      loop
+      bind __PILLAR__LOCAL__DNS__
+      forward . __PILLAR__CLUSTER__DNS__ {
+        force_tcp
+      }
+      prometheus :9253
+    }
+    ip6.arpa:53 {
+      errors
+      cache 30
+      reload
+      loop
+      bind __PILLAR__LOCAL__DNS__
+      forward . __PILLAR__CLUSTER__DNS__ {
+        force_tcp
+      }
+      prometheus :9253
+    }
+    .:53 {
+      errors
+      cache 30
+      reload
+      loop
+      bind __PILLAR__LOCAL__DNS__
+      forward . __PILLAR__UPSTREAM__SERVERS__ {
+        force_tcp
+      }
+      prometheus :9253
+    }
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: node-local-dns
+  namespace: kube-system
+  labels:
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+    kubeone.io/component: nodelocaldns
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: kube-dns-upstream
+  namespace: kube-system
+  labels:
+    k8s-app: kube-dns
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+    kubernetes.io/name: "KubeDNSUpstream"
+    kubeone.io/component: nodelocaldns
+spec:
+  ports:
+  - name: dns
+    port: 53
+    protocol: UDP
+    targetPort: 53
+  - name: dns-tcp
+    port: 53
+    protocol: TCP
+    targetPort: 53
+  selector:
+    k8s-app: kube-dns
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: node-local-dns
+  namespace: kube-system
+  labels:
+    k8s-app: node-local-dns
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
+    kubeone.io/component: nodelocaldns
+spec:
+  updateStrategy:
+    rollingUpdate:
+      maxUnavailable: 10%
+  selector:
+    matchLabels:
+      k8s-app: node-local-dns
+  template:
+    metadata:
+      labels:
+        k8s-app: node-local-dns
+    spec:
+      priorityClassName: system-node-critical
+      serviceAccountName: node-local-dns
+      hostNetwork: true
+      dnsPolicy: Default  # Don't use cluster DNS.
+      tolerations:
+      - operator: "Exists"
+      containers:
+      - name: node-cache
+        image: {{ Registry "k8s.gcr.io" }}/k8s-dns-node-cache:1.15.13
+        command:
+        - /bin/sh
+        - -c
+        - |-
+          sleep 10;
+          exec /node-cache -localip 169.254.20.10 -conf /etc/Corefile -upstreamsvc kube-dns-upstream
```

Comment:
The literal IP from here should be passed down as some parameter, based on resources package. Here and in few other places in this file